### PR TITLE
Fix typos plugin.rs

### PR DESCRIPTION
**Pull request by TheBlckbird**
# Objective

- There are multiple grammar mistakes in the `plugin.rs` file.

## Solution

- Corrects the grammar and spelling in the docs of `plugin.rs`

### CODE REVIEW

File: crates/bevy_app/src/plugin.rs [hunk 1]
```diff
@@ -25,8 +25,8 @@ pub trait Plugin: Downcast + Any + Send + Sync {
     /// Configures the [`App`] to which this plugin is added.
     fn build(&self, app: &mut App);
 
-    /// Has the plugin finished it's setup? This can be useful for plugins that needs something
-    /// asynchronous to happen before they can finish their setup, like renderer initialization.
+    /// This will be called when the plugin has finished its setup. Useful for plugins that need something
```
Comment (mockersf) [inline crates/bevy_app/src/plugin.rs:28]: this is not right, this is called to check if the plugin has finished its setup

Comment (TheBlckbird) [reply crates/bevy_app/src/plugin.rs:28]: Ok, I’ll just revert the changes

**Comment by TheBlckbird** [comment]
Like this?

**Comment by alice-i-cecile** [comment]
Merging as trivial, thank you <3

**Comment by TheBlckbird** [comment]
👍 :]

**Pull request merged**